contracts/core/Dispatcher.sol

```diff
@@ -150,6 +150,12 @@ contract Dispatcher is OwnableUpgradeable, UUPSUpgradeable, ReentrancyGuard, IDi
         string[] calldata connectionHops,
         string calldata counterpartyPortId
     ) external nonReentrant {
+        // We need to validate connectionHops & counterpartyPortId since they aren't validated in an internal
+        // function like all other instances of these arguments
+        if (connectionHops.length == 0 || bytes(counterpartyPortId).length == 0) {
```

Comment:
Should we check connectionHops length < 2? A single connection hop is also invalid from a vibc chain.

Reply:
+1

Reply:
yes that's valid, we can check for this instead

Reply:
code should be updated to check connecitonHops in all channel handshake methods